### PR TITLE
Improve release documentaiton

### PR DESCRIPTION
**Pull request by LecrisUT**
Adapted Keep-a-changelog format

### CODE REVIEW

**Comment by LecrisUT** [comment]
~~The [reference page](https://spglib--257.org.readthedocs.build/en/257/references.html) is broken? ~~ Fixed with 8d76e3cbf2f0356992afa280e716ec50b8773242

File: doc/README.md [hunk 1]
```diff
@@ -4,20 +4,25 @@ This directory contains python-sphinx documentation source.
 
 ## How to compile
 
-Python>=3.8 required for `importlib.metadata`.
+```shell
+$ pip install ".[docs]"
+$ python3 scripts/generate_changelog.py --type local
```
Comment (lan496) [inline doc/README.md]: `local` is not contained in choices for `--type`. Maybe adding it to choices is enough?

**Comment by lan496** [comment]
@LecrisUT Sorry for the late reply. Can you rebase this PR? I suspect the combination of Sphinx>6 and sphinx-bibtex<=2.5.0 may generate invalid HTML.

**Comment by LecrisUT** [comment]
> @LecrisUT Sorry for the late reply. Can you rebase this PR? I suspect the combination of Sphinx>6 and sphinx-bibtex<=2.5.0 may generate invalid HTML.

I have kinda fixed the reference by not having them numerically numbered. I forgot to check if cited references are properly linked though. I can try again with other sphinx-bibtex.

Also I have since got some more experience with how other projects manage their changelogs: [reference](https://github.com/Takishima/cmake-pre-commit-hooks/blob/main/CHANGELOG.md). I think that would be an easy format to have. We can use it for just the big-picture changes and it's less obscure than the yaml format here. What I didn't figure out yet though is how to update the fields when a new release is pushed. I can make it to push changelog update commits after the tagged release and have the changelog format as:
```markdown
## Current release
### Project related
- added my cat to the contributors list

## Version 3.4.10 (1970-01-01)
### Bug fixes
- my cat caught a big cockroach
```
Where the CI adds `## Version x.y.z (date)` jus below ,## Current release` after each release (excluding pre-releases).

Wdyt, sounds manageable?

**Comment by LecrisUT** [comment]
> @LecrisUT Sorry for the late reply. Can you rebase this PR? I suspect the combination of Sphinx>6 and sphinx-bibtex<=2.5.0 may generate invalid HTML.

~~Oh about this one. See #246, I will add that one as a dependency for this~~. Nevermind, that does not contain the fix I was thinking of, it is in this PR.

Ok, so I did some tests locally and it should be related to docutils  (see [this issue](https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/309)). It should be fixed in `docutils 0.20.1`, but it seems I need to check why rtd is failing dependencies (oh it's [book theme](https://github.com/executablebooks/sphinx-book-theme/issues/742) :()

**Comment by lan496** [comment]
Yeah, I'm also waiting for a newer version of sphinx-bibtex to avoid docutils issues for other projects...

**Comment by LecrisUT** [comment]
> Yeah, I'm also waiting for a newer version of sphinx-bibtex to avoid docutils issues for other projects...

Note that this issue is in `docutils` not `sphinx-bibtex`. You have to pull in `sphinx 7` that bumps `docutils` version to `0.20` otherwise this issue is not solved on the `sphinx-bibtex` side. On this side it's only `sphinx-book-theme` that is blocking us from upgrading to `sphinx 7`. Other themese like `furo` already support the latest version. Well technically there's `sphinx-design` as well, but that one is [easily fixed](https://github.com/executablebooks/sphinx-design/pull/131).

**Review by lan496 — APPROVED** [review]
Thank you for your hard work in supporting Sphinx 7 (https://github.com/executablebooks/sphinx-book-theme/pull/744)! LGTM for the contents.
I am not sure about the frequency of pre-releases in sphinx-book-theme, but it may be better to wait before we merge this PR?

**Comment by LecrisUT** [comment]
> LGTM for the contents.
> I am not sure about the frequency of pre-releases in sphinx-book-theme, but it may be better to wait before we merge this PR?

I think we should go forward with this to get the `changelog` format in and then build upon it. Then, before making an spglib release, we just revert 16a1efde2c89cc09bdd57775c85cb1fc541033a2 when upstream `sphinx-book-theme` has made a release. It should happen soon given the pressure on that package

**Comment by lan496** [comment]
I see. Then, I proceed to merge.

**Pull request merged**